### DISCUSSION TITLE
Removing entity changes reference or something?

### DISCUSSION THREAD

**Discussion by mkoskl** [doctrine/doctrine2#9889] [Support Questions] [answered]
Hello, 

I have a transaction and in it I first remove an entity (calling `$em->remove($entity)` and `$em->flush()`) and then get some properties of that entity (`$collection = $entity->getCollection()`). 

The entity should return (at least when used correctly) an array with only one entry but it returns like 4k which are not related to entity at all. Why would that be? 

I know it's not correct to use the entity after removing and flushing, that's bug in my code. But I was wondering why this happens? Is there a bug in Doctrine code too?

Doctrine ORM version I am currently using seems to be 2.8.1.

Br, Miika

**Comment by beberlei** [comment, marked as answer]
This looks like a bug, not sure its still in the latest but pretty sure it is.

PersistentCollection is triggered lazily but since id is null, the query conditions might be buggy

**Reply by mkoskl** [reply]
@beberlei I created issue on this and added some more information.
